packages/test/test-end-to-end-tests/src/test/nodeDidNotRunGc.spec.ts

```diff
@@ -0,0 +1,277 @@
+		disableSummary: boolean = true,
+		logger?: ITelemetryLogger,
+	): Promise<IContainer> => {
+		let summaryConfigOverrides: ISummaryConfiguration;
+		if (disableSummary) {
+			summaryConfigOverrides = { state: "disabled" };
+		} else {
+			const IdleDetectionTime = 100;
+			summaryConfigOverrides = {
+				...DefaultSummaryConfiguration,
+				...{
+					maxIdleTime: IdleDetectionTime,
+					maxTime: IdleDetectionTime * 12,
+					initialSummarizerDelayMs: 0,
+				},
+			};
+		}
+		const runtimeOptions: IContainerRuntimeOptions = {
+			summaryOptions: {
+				summaryConfigOverrides,
+			},
+		};
+		const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+			dataStoreFactory1,
+			registryStoreEntries,
+			undefined,
+			[innerRequestHandler],
+			runtimeOptions,
+		);
+		return provider.createContainer(runtimeFactory, { logger });
+	};
+
+	async function waitForSummaryOp(container: IContainer) {
+		await new Promise<void>((resolve) => {
+			container.deltaManager.on("op", (op: ISequencedDocumentMessage) => {
+				if (op.type === MessageType.Summarize) {
+					resolve();
+				}
+			});
+		});
+	}
+
+	beforeEach(async () => {
+		provider = getTestObjectProvider({ syncSummarizer: true });
+	});
+
+	itExpects(
+		"summarizeOnDemand should fail with NodeDidNotRunGC error",
+		[
+			{
+				eventName: "fluid:telemetry:SummarizerNode:NodeDidNotRunGC",
+				clientType: "noninteractive/summarizer",
+			},
+			{
+				eventName: "fluid:telemetry:Summarizer:Running:Summarize_cancel",
+				clientType: "noninteractive/summarizer",
+			},
+		],
+		async () => {
+			const mainContainer = await createContainer();
+			const mainDataStore = await requestFluidObject<TestDataObject1>(
+				mainContainer,
+				"default",
+			);
+			mainDataStore._root.set("mode", "write");
+			await waitForContainerConnection(mainContainer);
+
+			const { summarizer } = await createSummarizerFromFactory(
+				provider,
+				mainContainer,
+				dataStoreFactory1,
+				undefined,
+				undefined,
+				registryStoreEntries,
+			);
+
+			const result = summarizer.summarizeOnDemand({ reason: "test" });
+			const submitResult = await result.summarySubmitted;
+			assert(!submitResult.success, "The summary should have failed");
+			assert.strictEqual(
+				submitResult.stage,
+				"generate", // This would be "upload" without this change.
+				"Summary should have failed after generate",
+			);
+			assert.strictEqual(
+				submitResult.error.message,
+				"NodeDidNotRunGC",
+				"Summary failed with unexpected error",
+			);
+		},
+	);
+
+	itExpects(
+		"Regular summaries should pass on second attempt when NodeDidNotRunGC is hit",
+		[
+			{
+				eventName: "fluid:telemetry:SummarizerNode:NodeDidNotRunGC",
+				clientType: "noninteractive/summarizer",
+			},
+			{
+				eventName: "fluid:telemetry:Summarizer:Running:Summarize_cancel",
+				clientType: "noninteractive/summarizer",
+			},
+			{
+				eventName: "fluid:telemetry:Summarizer:Running:gcUnknownOutboundReferences",
```

Comment:
It's too late on Friday for me to figure out where this is coming from.  Is it a concern?

Reply:
@agarwal-navin can you explain why and maybe put a comment?

Reply:
Added comment, take a look.

Reply:
That's right, I remember now, thanks!